File: java/src/main/java/com/a65apps/kalimullinilnazrafilovich/interactors/notification/NotificationRepository.java
package com.a65apps.kalimullinilnazrafilovich.interactors.notification;

import com.a65apps.kalimullinilnazrafilovich.entities.BirthdayNotification;
import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo;

import java.util.GregorianCalendar;

public interface NotificationRepository {
    BirthdayNotification setBirthdayReminder(ContactDetailsInfo contactDetailsInfo,
                                             GregorianCalendar gregorianCalendar);

    BirthdayNotification removeBirthdayReminder(ContactDetailsInfo contactDetailsInfo);

    BirthdayNotification getBirthdayNotificationEntity(ContactDetailsInfo contactDetailsInfo,
                                                       GregorianCalendar gregorianCalendar);

}

File: java/src/main/java/com/a65apps/kalimullinilnazrafilovich/interactors/notification/NotificationRepository.kt
package com.a65apps.kalimullinilnazrafilovich.interactors.notification

import com.a65apps.kalimullinilnazrafilovich.entities.BirthdayNotification
import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo
import java.util.*

interface NotificationRepository {
    fun setBirthdayReminder(contactDetailsInfo: ContactDetailsInfo,
                            gregorianCalendar: GregorianCalendar): BirthdayNotification

    fun removeBirthdayReminder(contactDetailsInfo: ContactDetailsInfo): BirthdayNotification

    fun getBirthdayNotificationEntity(contactDetailsInfo: ContactDetailsInfo,
                                      gregorianCalendar: GregorianCalendar?): BirthdayNotification


}